backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/RuntimeNames.kt

```diff
@@ -11,4 +11,5 @@ object RuntimeNames {
     val independent = FqName("kotlin.native.internal.Independent")
     val filterExceptions = FqName("kotlin.native.internal.FilterExceptions")
     val kotlinNativeInternalPackageName = FqName.fromSegments(listOf("kotlin", "native", "internal"))
+    val associatedObjectKey = FqName("kotlin.reflect.AssociatedObjectKey")
```

Comment:
Maybe some other package, like `kotlin.annotations`?

Reply:
I like the idea of having both ends of this machinery (`@AssociatedObjectKey` and `getAssociatedObject()`) in the same package. Not sure if this makes much sense though.

Reply:
`kotlin.annotation` package is for general annotation facility support types. This annotation is specialized for reflection use cases, so its `kotlin.reflect` package placement looks logical.